client/src/www/app/outline_server_repository/server.ts

```diff
@@ -88,7 +88,10 @@ export class OutlineServer implements Server {
   async connect() {
     let tunnelConfig: TunnelConfigJson;
     if (this.type === ServerType.DYNAMIC_CONNECTION) {
-      tunnelConfig = await fetchTunnelConfig(this.tunnelConfigLocation);
+      tunnelConfig = await fetchTunnelConfig(
+        this.vpnApi,
```

Comment:
Pass a fetcher instead.

And inject the fetcher in the Server constructor.

Reply:
Done.